python/tank/platform/application.py

```diff
@@ -265,36 +265,25 @@ def log_exception(self, msg):
     ##########################################################################################
     # internal API
 
-    def log_metric(self, action, log_version=False, log_once=False):
-        """Logs an app metric.
-
-        :param action: Action string to log, e.g. 'Execute Action'
-        :param log_version: If True, also log a user attribute metric for the
-            version of the app. Default is `False`.
-        :param bool log_once: ``True`` if this metric should be ignored if it
-            has already been logged. Defaults to ``False``.
-
-        Logs a user activity metric as performed within an app. This is a
-        convenience method that auto-populates the module portion of
-        `tank.util.log_user_activity_metric()`.
-
-        If the optional `log_version` flag is set to True, this method will
-        also log a user attribute metric for the current version of the app.
-
-        Internal Use Only - We provide no guarantees that this method
-        will be backwards compatible.
-
+    @property
+    def _metric_properties(self):
+        """
+        :returns: A dictionary with metric properties specific to this application.
         """
-        # the action contains the engine and app name, e.g.
-        # module: tk-multi-loader2
-        # action: (tk-maya) tk-multi-loader2 - Load...
-        full_action = "(%s) %s %s" % (self.engine.name, self.name, action)
-        log_user_activity_metric(self.name, full_action, log_once=log_once)
+        # Always create a new dictionary so the caller can safely modify it.
+        return {
+            EventMetric.KEY_APP: self.name,
+            EventMetric.KEY_APP_VERSION: self.version
+        }
 
-        if log_version:
-            # log the app version as a user attribute
-            log_user_attribute_metric(
-                "%s version" % (self.name,), self.version, log_once=log_once)
+    def _get_metrics_context(self):
```

Comment:
`context` has a generally different meaning in tk - we should probably rename it to `_get_metrics_properties` or something like that to avoid confusion. This method seems very similar to `_metric_properties` - do we need both? Reading the docstring doesn't really help me understand the difference: "A dictionary with metric properties specific to this application." vs "A dictionary with properties to use when emitting a metric event for this application in the current engine." Again maybe examples will help .

Reply:
The idea was to have something which would return needed properties at any level (Engine/App/Framework), taking into account the hierarchy, hence the '_context' and something which would return the properties for a given Engine/App/Framework, without taking the hierarchy into account. Revisiting that, as it seems to be overkill to have both.